hashring/hashring.go

```diff
@@ -187,81 +181,72 @@ func (r *HashRing) AddRemoveServers(add []string, remove []string) bool {
 	return changed
 }
 
-// hasServer returns true if the server exists in the ring, false otherwise
+// HasServer returns true if the server exists in the ring, false otherwise
 func (r *HashRing) HasServer(address string) bool {
 	r.servers.RLock()
-	server := r.servers.byAddress[address]
-	r.servers.RUnlock()
-
-	return server
+	defer r.servers.RUnlock()
+	return r.servers.byAddress[address]
```

Comment:
The right way to test membership of a map is:

```go
value, ok := r.servers.byAddress[address]
```

`ok` is false if the entry was not in the map. If there is a zero value in the map you will not see the difference in `value`, but you would in `ok`